nevergrad/functions/powersystems/core.py

```diff
@@ -0,0 +1,193 @@
+#
+#losses = []
+#for i in range(optim_steps):
+#    candidate = optimizer.ask()
+#    v, hydro_prod, hydro_prod_per_ts, consumption_per_ts = simulate_power_system(candidate.data)
+#    losses += [min(v, min([float("Inf")] + losses))]
+#    optimizer.tell(candidate, v)
+#
+#ax = plt.subplot(1, 2, 1)
+#ax.set_xlabel('iteration number')
+#ax.plot(losses, label='losses') 
+#ax = plt.subplot(1, 2, 2)
+#ax.plot(np.linspace(0,1,len(consumption_per_ts)), consumption_per_ts, label='consumption')
+#ax.plot(np.linspace(0,1,len(hydro_prod_per_ts)), hydro_prod_per_ts, label='hydro')
+#for i in range(N):
+#    hydro_ts = [hydro_prod_per_ts[j] for j in range(i, len(hydro_prod_per_ts), N)]
+#    ax.plot(np.linspace(0,1,len(hydro_ts)), hydro_ts, label='dam ' + str(i) + ' prod')
+#ax.set_xlabel('time step')
+#ax.set_ylabel('production per ts')
+#ax.legend() #(l1, l2))
+#plt.show()
+#plt.savefig("ps.png")
+#plt.waitforbuttonpress()
+#recommendation = optimizer.recommend()  #minimize(square)
+#print(recommendation)  # optimal args and kwargs
+
+
+
+
+
+# Real life is more complicated! This is a very simple model.
+class Agent():
+    """An agent has an input size, an output size, a number of layers, a width of its internal layers 
+    (a.k.a number of neurons per hidden layer)."""
+    def __init__(self, input_size, output_size, layers=2, layer_width=20):
+        assert(layers >= 2)
+        self.input_size = input_size
+        self.output_size = output_size
+        self.layers = []
+        self.layers += [np.random.rand(input_size, layer_width)]
+        for i in range(layers-2):
+            self.layers += [np.random.rand(layer_width, layer_width)]
+        self.layers += [np.random.rand(layer_width, output_size)]
+        assert len(self.layers) == layers
+
+    def GetParamNumbers(self):
+        return sum([np.prod(l.shape) for l in self.layers])
+
+    def SetParams(self, ww):
+        w = [w for w in ww]
+        assert(len(w) == self.GetParamNumbers())
+        for l in self.layers:
+            s = np.prod(l.shape)
+            l = np.reshape(np.array(w[:s]), l.shape)
+            w = w[s:]
+
+    def GetOutput(self, inp):
+        output = np.array(inp).reshape(1, len(inp))
+        for l in [self.layers[i] for i in range(len(self.layers) - 1)]:
+            output = np.tanh(np.matmul(output, l))
+        return np.matmul(output, self.layers[-1])
+
+
+
+
+class PowerSystemVariable(Variable):
+
+    def __init__(self, num_stocks:int, depth: int, width: int, dimension: int) -> None:
+        
+        # Simple instrumentation: just the number of params.
+        super().__init__()
+        self.num_stocks = num_stocks
+        self.depth = depth
+        self.width = width
+        self._specs.update(dimension=dimension)
+
+    # pylint: disable=unused-argument
+    def _data_to_arguments(self, data: np.ndarray, deterministic: bool = True) -> Any:
+        assert len(data) == self.dimension
+        return data, {}
```

Comment:
instrumentation is tricky... I'll try to make it simpler but there is soooo much involved that I keep failing :s 
here I think what you would like is actually to return `(data,), {}` which may solve the tests
then again for now this variable is not doing anything :s so you could go with an `Array` variable directly

Reply:
done!